x/gamm/keeper/pool.go

```diff
@@ -295,3 +297,10 @@ func convertToCFMMPool(pool poolmanagertypes.PoolI) (types.CFMMPoolI, error) {
 	}
 	return cfmmPool, nil
 }
+
+func GetPoolIdFromSharesDenom(denom string) (uint64, error) {
```

Comment:
does this have to be exported?

Also, please godoc it

Reply:
unexported and added godoc

https://github.com/osmosis-labs/osmosis/pull/4051/commits/7dfa7d09b58891e7f4598d85a572d8aeb0e6f87f